src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -396,6 +449,113 @@ void Instance::HandleAddOrUpdateWiFiNetwork(HandlerContext & ctx, const Commands
     }
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+void Instance::HandleAddOrUpdateWiFiNetworkWithPDC(HandlerContext & ctx,
+                                                   const Commands::AddOrUpdateWiFiNetwork::DecodableType & req)
+{
+    // Credentials must be empty when configuring for PDC, it's only present to keep the command shape compatible.
+    if (!req.credentials.empty())
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand, "credentials");
+        return;
+    }
+
+    auto && networkIdentity = req.networkIdentity.Value(); // presence checked by caller
+    if (networkIdentity.size() > kMaxCHIPCompactNetworkIdentityLength ||
+        Credentials::ValidateChipNetworkIdentity(networkIdentity) != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand, "networkIdentity");
+        return;
+    }
+
+    if (req.clientIdentifier.HasValue() && req.clientIdentifier.Value().size() != CertificateKeyId::size())
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand, "clientIdentifier");
+        return;
+    }
+
+    bool provePossession = req.possessionNonce.HasValue();
+    if (provePossession && req.possessionNonce.Value().size() != kPossessionNonceSize)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand, "possessionNonce");
+        return;
+    }
+
+    auto status = Protocols::InteractionModel::Status::Failure;
+    auto driver = mpDriver.Get<WiFiDriver *>();
+    {
+        // If the client is requesting re-used of a Client Identity, find the existing network it belongs to
+        Optional<uint8_t> clientIdentityNetworkIndex;
+        if (req.clientIdentifier.HasValue())
+        {
+            CertificateKeyId clientIdentifier(req.clientIdentifier.Value().data());
+            uint8_t networkIndex = 0;
+            EnumerateAndRelease(driver->GetNetworks(), [&](const Network & network) {
+                if (network.clientIdentifier.HasValue() && clientIdentifier.data_equal(network.clientIdentifier.Value()))
+                {
+                    clientIdentityNetworkIndex.SetValue(networkIndex);
+                    return Loop::Break;
+                }
+                networkIndex++;
+                return Loop::Continue;
+            });
+            if (!clientIdentityNetworkIndex.HasValue())
+            {
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand,
```

Comment:
Spec issue: spec should define that if the ClientIdentifier does not match an existing network the error is INVALID_COMMAND.

Reply:
Yes, or alternatively we could define a new `Status` enum value... It seems the convention is that we use `INVALID_COMMAND` when the input does not conform to the constraints (e.g. in the non-PDC case when the credentials field doesn't follow one of the allowed formats in terms of lengths and hex vs ascii). Having a syntactically valid value that doesn't match some existing data should maybe be a different error code, maybe `NOT_FOUND` ? But I'm unsure if this should be an IM status code or a NetworkConfigResponse Status.

Reply:
Going with `NOT_FOUND`

Reply:
That's fine, but we need a spec issue (labeled "SDK Feedback") about this not being defined.

Reply:
https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/8534